content/source/docs/enterprise/vcs/troubleshooting.html.md

```diff
@@ -0,0 +1,86 @@
+
+TFE uses OAuth 1 to authenticate the user to Bitbucket Server. The first step in the authentication process is for TFE to call Bitbucket Server to obtain a "request token". After the call completes TFE will redirect you to Bitbucket Server with the request token.
+
+An error occurs when TFE calls to Bitbucket Server to obtain the request token but the request is rejected. Some common reasons for the request to be rejected are:
+
+- The API endpoint is unreachable; this can happen if the address or port is incorrect or the domain name doesn't resolve.
+- The certificate used on Bitbucket Server is rejected by the TFE HTTP client because the SSL verification fails. This is often the case with self-signed certificates or when the server with TFE is not configured to trust the signing chain of the Bitbucket Server SSL certificate.
+
+To fix this issue, do the following:
+
+- Verify that the instance running Terraform Enterprise can resolve the domain name and can reach Bitbucket Server.
+- Verify that the TFE client accepts the HTTPS connection to Bitbucket Server. This can be done by performing a `curl` from the TFE instance to Bitbucket Server; it should not return any SSL errors.
+- Verify that the Consumer Key, Consumer Name, and the Public Key are configured properly in Bitbucket Server.
+- Verify that the HTTP URL and API URL in TFE are correct for your Bitbucket Server instance. This includes the proper scheme (HTTP vs HTTPS), as well as the port.
+
+### Creating a workspace from a repository hangs indefinitely, displaying a spinner on the confirm button
+
+If you were able to connect TFE to Bitbucket Server but cannot create workspaces, it often means TFE isn't able to automatically add webhook URLs for that repository.
+
+To fix this issue:
+
+- Make sure you haven't manually entered any webhook URLs for the affected repository or project. Although the Bitbucket Web Post Hooks Plugin documentation describes how to manually enter a hook URL, TFE handles this automatically. Manually entered URLs can sometimes interfere with TFE's operation.
+
+    To check the hook URLs for a repository, go to the repository's settings, then go to the "Hooks" page (in the "Workflow" section) and click on the "Post-Receive WebHooks" link.
+
+    Also note that some Bitbucket Server versions might allow you to set per-project or server-wide hook URLs in addition to per-repository hooks. These should all be empty; if you set a hook URL that might affect more than one repo when installing the plugin, go back and delete it.
+- Make sure you aren't trying to connect too many workspaces to a single repository. Bitbucket Server's webhooks plugin can only attach five hooks to a given repo, and TFE can have problems if you exceed the limit. You might need to create additional repositories if you need to make more than five workspaces from a single configuration repo.
```

Comment:
"TFE can have problems..." sounds a little strange here since we're already in a specific problem - is this clause even needed?